src/test/ui/self/arbitrary_self_types_pin_lifetime_mismatch.stderr

```diff
@@ -5,6 +5,12 @@ LL |     fn a(self: Pin<&Foo>, f: &Foo) -> &Foo { f }
    |                              ----     ----   ^ ...but data from `f` is returned here
    |                              |
    |                              this parameter and the return type are declared with different lifetimes...
+   |
+   = note: each elided lifetime in input position becomes a distinct lifetime
+help: consider introducing a named lifetime parameter
+   |
+LL |     fn a<'a>(self: Pin<&'a Foo>, f: &'a Foo) -> &Foo { f }
+   |         ++++            ++           ++
```

Comment:
I'm somewhat surprised by the suggestions I'm seeing that we don't also suggest adding `'a` to the return type. Maybe an existing bug? No need to handle it in this case directly, but would we should make a note of it.

Reply:
Do you want me to add a comment about that note?

I'm not 100% sure it's a bug. Seems like the code that added the hint did not take into consideration return types. This diagnosis error in this module is implemented very differently than what is in [add_missing_lifetime_specifiers_label](https://github.com/rust-lang/rust/blob/f60513ec8d97414a346cedb755a4cac3abd55ed1/src/librustc_resolve/late/diagnostics.rs#L1080-L1209) which does add lifetime hints for the return type.

Reply:
Let's just create a targeted ticket for this instead of leaving it in the code. It'd be `p-low`, as we [already have follow up suggestions](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bdc1da742b2f69195f015f8eed76f7db) to get you to working code, although [it isn't great when there's another lifetime involved](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c58b8cd5ef41f9fbc49b9a1d226534dc)

Reply:
@estebank I created #94492 and made the changes you suggested.